sql-ballerina/src/sql/connection-pool.bal

```diff
@@ -48,7 +48,7 @@ class GlobalConnectionPoolContainer {
     }
 
     public isolated function getGlobalConnectionPool() returns ConnectionPool {
-        return self.connectionPool;
+        return self.connectionPool.cloneReadOnly();
```

Comment:
Works without this too, right? When a class is marked as `readonly`, the fields also become immutable.

Reply:
Fixed in 5eb2441